operator/internal/controller/konflux_controller.go

```diff
@@ -401,10 +409,11 @@ func (r *KonfluxReconciler) applyKonfluxReleaseService(ctx context.Context, owne
 // applyKonfluxUI creates or updates the KonfluxUI CR.
 func (r *KonfluxReconciler) applyKonfluxUI(ctx context.Context, owner *konfluxv1alpha1.Konflux) error {
 	log := logf.FromContext(ctx)
-
 	var spec konfluxv1alpha1.KonfluxUISpec
 	if owner.Spec.KonfluxUI != nil && owner.Spec.KonfluxUI.Spec != nil {
 		spec = *owner.Spec.KonfluxUI.Spec
+	} else {
```

Comment:
you can remove the else. `var spec konfluxv1alpha1.KonfluxUISpec` create a new struct with zero values. the issue we was with the omitempty annotation.

Reply:
Done